communication/src/dtls_message_channel.cpp

```diff
@@ -94,9 +87,19 @@ auto SessionPersist::restore(mbedtls_ssl_context* context, bool renegotiate, uin
 		return NO_SESSION;
 	}
 
+    increment_use_count();
+    save(saver);
```

Comment:
We should probably check if the session is expired first and then increment the counter (or change the implementation of `has_expired()`), otherwise the session will be invalidated after 2 attempts instead of 3 (not sure if that was the intent)